rfcs/rfc-using-global-store-for-form-components.md

```diff
@@ -0,0 +1,317 @@
+        store[prop] = { ...store[prop], ...value }
+      } else {
+        store[prop] = value
+      }
+      return
+    }
+    this.setProp(store[prop], nestedProperty, value)
+  }
+```
+
+We envision expanding the low-level components like `SingleCheckbox.vue` to include component
+to display radio buttons, text fields and date fields etc.  Each of these components would talk
+directly to the store using the store getter `getNestedAttribute` and the mutator `setNestedAttribute`.
+
+The new approach has more advantages as the forms get more complex and have more nested objects.  For
+example:
+
+```html
+<template>
+  <FormRow>
+    <InLineCheckBox id="corp_owner" :path="path" :option="true">
+        Owned by corporate entity
+    </InLineCheckBox>
+  </FormRow>
+  <!-- corporation sub-component start -->
+  <FormRow v-if="getNestedAttribute(path, 'corp_owner') === true">
+    <TextField id="name" :path="path + '.corporation'" rules="max:40">
+      Corporation Name</TextField>
+  </FormRow>
+  <!-- corporation sub-component start -->
+  <!-- individual sub-component start -->
+  <FormRow v-if="getNestedAttribute(path, 'corp_owner') !== true">
+    <TextField id="last_name" :path="path + '.individual'" rules="max:20">
+      Owner's Last Name</TextField>
+    <TextField id="first_name" :path="path + '.individual'" rules="max:20">
+      Owner's First Name</TextField>
+  </FormRow>
+  <!-- individual sub-component end -->
+  <FormRow>
+    <TextField id="address" :path="path" placeholder="Address" rules="max:25">
+        Address Line
+    </TextField>
+  </FormRow>
+  <FormRow>
+    <TextField id="city" :path=path rules="max:20">City</TextField>
+    <ProvinceField id="province" :path=path>Prov / State</ProvinceField>
+    <TextField id="postal" :path=path>Postal / Zip</TextField>
+    <PhoneField id="phone" :path=path rules="phone">Phone</PhoneField>
+  </FormRow>
+</template>
+```
+
+
+# Drawbacks
+
+We don't see this proposal as having significant drawbacks. If adopted, this new
+approach could be implemented gradually and could be used alongside the current approach.
+
+Some developers may have concerns about moving the business logic from components to the store,
+but we suggest using Pinia for business logic is better and more flexible than storing 
+business logic in components. We'd be interested in what the group's feedback might be on this topic.
+
+Form validation is a big part of developing reusable form components. We haven't mentioned form
+validation in this RFC, but would like to do so in a future RFC.
+
+# Alternatives
+
+We could use the Composition API instead of Pinia. The Composition API is native to Vue and has
+the advantage of being lighter weight and more flexible. However, with Pinia the state is viewable
+by developers using the Vue Devtools plugin, whereas shared state is not viewable when using the 
+Composition API.
+
+Also, creating sharable state using standard Vue doesn't offer the same safe guards as Pinia.
```

Comment:
This is a good point, but its a lot less boiler plate, and you can protect your global variables quite easily if you want to, by returning computed vars for any read only vars you want